### PR TITLE
ci: Use ruff formatter

### LINKED ISSUE #1658: [ENHANCEMENT] Replace `black` with `ruff` formatter
**Is your feature request related to a problem? Please describe.**
We are currently using two linters/formatters `Black` and `ruff`. They sometimes don't play nice.

**Describe the solution you'd like**
`ruff` has [recently release](https://twitter.com/charliermarsh/status/1716861072374657497) a formatter that claims nearly-complete Black compatibility. I propose we switch our CI style check to only use `ruff`, formatting should be much more consistent with a single tool.

**Describe alternatives you've considered**
We all have been using multiple formatters for the past few years given the Python linting/formatting ecosystem. Usually some combination of `black` and `isort` and more recently `ruff`. Now that `ruff` has a full formatter let's consider switching to a single tool.

### PR DESCRIPTION
**Pull request by anticorrelator**
resolves #1658 

Simplify CI toolchain by removing `black` in favor of `ruff format`

With the release of [ruff formatter](https://github.com/astral-sh/ruff/releases/tag/v0.1.3), we can use ruff for both formatting and linting, improving consistency and issues with black and ruff disagreeing with eachother.

For now I'm leaving out the jupyter notebook extension, since reformatting our notebooks will lead to a significant number of changes.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -178,12 +175,10 @@ check = [
 
 [tool.hatch.envs.style.scripts]
 check = [
-  "black --check --diff --color .",
   "ruff .",
 ]
 fix = [
-  "black .",
-  "ruff --fix .",
+  "ruff format .",
```
Comment (RogerHYang) [inline pyproject.toml:181]: fix is still needed
```suggestion
  "ruff format .",
  "ruff --fix .",
```

File: pyproject.toml [hunk 2]
```diff
@@ -178,12 +175,10 @@ check = [
 
 [tool.hatch.envs.style.scripts]
 check = [
-  "black --check --diff --color .",
   "ruff .",
```
Comment (RogerHYang) [inline pyproject.toml:178]: still needs to check format
```suggestion
  "ruff format --check --diff .",
  "ruff .",
```

Comment (anticorrelator) [reply pyproject.toml:178]: thanks, got it

File: examples/dolly-pythia-fine-tuned/create_data_with_dolly_and_pythia.ipynb [hunk 3]
```diff
@@ -41,33 +41,31 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "openai.api_key = \"your key here\"\n",
+    "assert openai.api_key != \"your key here\", \"Set your key\""
+   ]
+  },
```
Comment (mikeldking) [inline examples/dolly-pythia-fine-tuned/create_data_with_dolly_and_pythia.ipynb:68]: if adding this let's use getpass instead.

File: pyproject.toml [hunk 4]
```diff
@@ -120,9 +119,7 @@ dependencies = [
 [tool.hatch.envs.style]
 detached = true
 dependencies = [
-  "black~=23.3.0",
-  "black[jupyter]~=23.3.0",
-  "ruff~=0.0.290",
+  "ruff~=0.1.3",
```
Comment (mikeldking) [inline pyproject.toml:122]: nice

**Review by mikeldking — APPROVED** [review]

File: examples/dolly-pythia-fine-tuned/create_data_with_dolly_and_pythia.ipynb (hunk 3 above)
Comment (anticorrelator) [reply examples/dolly-pythia-fine-tuned/create_data_with_dolly_and_pythia.ipynb:68]: This isn't added, I just moved a cell around to fix a linting issue

File: pyproject.toml [hunk 5]
```diff
@@ -178,12 +175,12 @@ check = [
...
 [tool.hatch.envs.style.scripts]
 check = [
-  "black --check --diff --color .",
   "ruff .",
+  "ruff format --check --diff .",
 ]
 fix = [
-  "black .",
   "ruff --fix .",
+  "ruff format .",
```
Comment (RogerHYang) [inline pyproject.toml:183]: I think in general format should go first before `fix`. For example, in the future, format may be able to re-format lines that are too long but `fix` may not be able to.

**Pull request merged**